src/ucp/core/ucp_ep.h

```diff
@@ -299,8 +296,30 @@ typedef struct {
         ucs_queue_head_t          match_q;       /* Queue of receive data or requests,
                                                     depends on UCP_EP_FLAG_STREAM_HAS_DATA */
     } stream;
+
+    struct {
+        uintptr_t                 dest_ep_ptr;   /* Remote EP pointer */
```

Comment:
why is this moved to ext_proto? it means we would have to create ext_proto for hpc too

Reply:
because `listener` and `user_data` from generic extension can co-exist now, they were in union before. ep and generic ext are full with cache line size.

Reply:
need to refactor this so hpc use case will not use more memory

Reply:
No way without on demand allocation per endpoint (this is why we added EP allocator) or backward compatibility break (we can decide which extensions to use only at ucp_worker creation, for example, if we document that listener can be created only if wakeup feature (or some other) was requested).

Reply:
so need to move other fields around, like user_data or err_cb

Reply:
they don't require special feature and can be requested per EP, so we must create the extension anyway.

Reply:
ok, so pls find other way to avoid memory size increase

Reply:
OK, will add malloc/free for on demand fields.

Reply:
i would add "non fast path" malloc'ed extension with listener and err_cb (both not used in HPC, user_data is used on fast path in stream), we need to move at least 2 pointers to have a space for this.
union of listener with ep_match should work from first view but it looks not safety for me.

Reply:
or modify strided_alloc to be have blocks for different stride counts
so would not have to define up front how many strides are extensions per ep